src/adapters/proposals-query.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017 The "mysteriumnetwork/mysterium-vpn" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// @flow
+import type { HttpQueryParams } from './interface'
+
+class ProposalsQuery {
+  fetchConnectCounts: boolean = true
+  providerId: ?string
+
+  constructor (fetchConnectCounts: ?boolean, providerId: ?string) {
+    if (typeof fetchConnectCounts === 'boolean') {
```

Comment:
It would help is we see full use case of library:

A. Either caller fetches proposals with Tequilapi defaults
```
findProposals()
```

B. Either caller fetches proposals with customisation and then he/she needs to how everything how to customise
*B1 - Magic order*
```
proposals = findProposals(new ProposalsQuery(true))
proposals = findProposals(new ProposalsQuery(10)) <-- double-types looks weird 
```

*B2 - Clear parameter order*
```
proposals = findProposals(new ProposalsQuery(true, undefined)) <-- looks weird
proposals = findProposals(new ProposalsQuery(true)) <-- better
```

*B3: default struct & setters*
```
query = new ProposalsQuery() <-- default struct
query.fetchConnectCounts = true <-- customisation
proposals = findProposals(query)
```

*B4: named parameters*
proposals = findProposals(new ProposalsQuery({fetchConnectCounts: true}))

query = new ProposalsQuery({fetchConnectCounts: true})
proposals = findProposals(query)

Reply:
My vote is towards B3 or B4